### ISSUE TITLE
Tooltip a11y review: Re-consider default value for triggerAriaAttribute

### ISSUE THREAD

**Issue by behowell** [microsoft/fluentui#21297]
According to the tooltip ARIA documentation (linked below), the tooltip should be the `aria-describedby` value for element that it targets. However, we currently default to `aria-label`. It could be confusing or unexpected if adding a tooltip to a control changes its label for screen readers.

References:
* https://www.w3.org/TR/wai-aria-1.1/#tooltip
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role

**Comment by jurokapsiar** [comment]
The default was proposed because most common reason for adding a tooltip is to provide a label to an icon-only button. From my experience with reviewing product PRs, about 80% of usages of Tooltip are for providing a label.

**Comment by behowell** [comment]
@jurokapsiar The primary concern is that setting aria-label is not a safe default, since it overrides the label of the trigger element. It's the correct default for icon buttons, but it's unexpected for other uses, and could result in accessibility issues by overriding the control's actual label. Setting aria-describedby seems like a safer default, as it is still read by (most?) screen readers in their default configuration (please correct me if I'm wrong). It'll still be necessary to change it to aria-label for icon buttons, but if that's not done, the text should at least still be read as the description. And for all other controls, their labels won't be unexpectedly overridden by adding a tooltip.

The secondary concern is that the docs linked in the issue description all mention aria-describedby, and not aria-label. It seems to be the recommended behavior for tooltips.

cc @smhigley

**Comment by jurokapsiar** [comment]
>  is still read by (most?) screen readers

The label is not used only for narration. It is also present in the list of interactive elements. In addition to that, voice control uses the label to navigate

If we make description the default, what is our way of informing the developers that a label needs to be provided? Are there any build time or runtime checks that we can provide? Making buttons without text really ugly? Accessibility checkers can discover that, but from our experience, these are being added to automation quite late in the development process or even when features get to production.

I understand description might be more expected default for accessibility experts, but with going this way, we will be making it harder to achieve common scenarios and easier to achieve scenarios that are less frequent.

**Comment by behowell** [comment]
Based on an offline discussion, we decided to have no default, and make `relationship` a required prop.
